<a href="https://colab.research.google.com/github/wantfree8937/Machine/blob/main/%EC%BB%B4%EA%B3%B5%EC%9E%85%EB%AC%B8%EA%B3%BC%ED%8C%8C%EC%9D%B4%EC%8D%AC%EA%B3%BC%EC%A0%9C1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# (실습) 프로그래밍 시작하기

**주의 사항**

* 기존에 작성된 코드셀/텍스트셀은 수정하지 않는다.
* 필요한 경우 코드셀 또는 텍스트셀을 추가할 수 있다.
* 실습 파일을 구글 드라이브에서 링크로 누구가 공유할 수 있도록 설정하세요.

**문제 1** 

아래 명령문들을 실행해서 오류가 발생하는 경우 오류를 설명하고 수정하라.

```python
print("Hello World"
print("Hello World)
print('Hello World")
print(+2)
print(2++2)
print(023)
print(21 8)
```

**문제1 답안**

```python
print("Hello World"
# print("Hello World" -> print("Hello World"), 소괄호누락
```

```python
print("Hello World)
# print("Hello World) -> print("Hello World"), 큰따옴표누락
```

```python
print('Hello World")
# print('Hello World") -> print("Hello World"), 큰따옴표 or 작은따옴표로 통일해야함
```

```python
print(023)
# print(023) -> print(23), 0이 맨 앞에 왔음
```



In [ ]:
print(23)

23



```python
print(21 8)
# print(21 8) -> print(21,8), 쉼표 누락
```



In [ ]:
print(21,8)

21 8


**문제 2** 

변수 선언과 할당 관련 주의해야할 점들을 다룬다.
아래 명령문들을 실행해서 오류가 발생하는 경우 오류를 설명하고 수정하라.

```python
23 = n
x = y = 1
x = 2;
y = x + 1.
z = x y
```

**문제2 답안**



```python
23 = n 
# 23 = n -> n = 23, 리터럴에 변수 할당x

```





```python
z = x y
# z = x y -> z = x*y, 곱셈 누락
```



**문제 3**

연산자를 이용해 수식으로 표현해보고 그 결과를 확인하여라.

* 정수 25와 정수 12를 더하시오.
* 정수 13에서 정수 19를 빼시오.
* 정수 92과 부동소수점 8.0 을 곱하시오.
* 부동소수점 81.5을 부동소수점 2.0으로 나누시오.
* 정수 7를 5로 나눴을 때, 몫을 구하시오.
* 정수 100을 정수 7로 나눈 나머지를 구하시오.
* 정수 3의 세제곱을 구하시오.

**문제3 답안**

In [ ]:
25 + 12

37

In [ ]:
13 - 19

-6

In [ ]:
92 * 8.0

736.0

In [ ]:
81.5 / 2.0

40.75

In [ ]:
7 // 5

1

In [ ]:
100 % 7

2

In [ ]:
3 ** 3

27

**문제 4**

다음 각각의 자료형을 확인하는 코드를 작성하라.

* `3`
* `3.0`
* `'3'`
* `'3.0'`
* `"3"`
* `"3.0"`

**문제4 답안**

In [ ]:
type(3)

int

In [ ]:
type(3.0)

float

In [ ]:
type('3')

str

In [ ]:
type('3.0')

str

In [ ]:
type("3")

str

In [ ]:
type("3.0")

str

**문제 5** 

아래와 같이 변수들이 선언되었다고 가정하자.

```python
width = 17
height = 12.0
delimiter = '.'
```

다음에 오는 각각의 표현식들에 대해, 표현식의 값과 자료형을 확인하라.

```python
width//2
width/2
width/2.0
height/3
1+2*5
delimiter*5
```

**문제5 답안**

In [ ]:
width = 17
height = 12.0
delimiter = '.'

In [ ]:
print(width//2)
print(width/2)
print(width/2.0)
print(height/3)
print(1+2*5)
print(delimiter*5)

8
8.5
8.5
4.0
11
.....


**문제 6**

1$\text{m}^2$는 0.3025평이다. 
84$\text{m}^2$는 몇 평인가? 
반대로 30평은 몇 $\text{m}^2$인가?

**문제6 답안**

In [ ]:
a = 0.3025 # a = 1m^2
print(a * 84)

25.41


25.41평

In [ ]:
(10000 / 3025) * a

1.0

In [ ]:
10000 / 3025 * 30

99.17355371900827

99.17$\text{m}^2$


**문제 7**

우유 가격은 820원이고, 아이스크림 가격은 1500원이다.
아이스크림은 3개이상 구입시 5%를 할인해준다.
우유 2개와 아이스크림 3개를 구입할 때 지불해야 하는 가격을 변수를 활용하여 코드로 작성하여라.

힌트: 변수를 이용한다.

**문제7 답안**

In [ ]:
milk = 820
ice = 1500
(2 * milk) + (3 * ice) - (5 / 100 * 3 * ice)

5915.0

5915원

**문제 8**

100만원을 월이율 3%인 예금에 1년간 복리로 은행에 맡겼다.
1년 후에 받는 돈은 얼마인지를 확인하는 코드를 작성하여라.
단, 다음 형식으로 출력하라.

> 1년 후 받는 원금 + 이자는 OOO원입니다.

힌트: 변수를 이용한다.

**문제8 답안**

In [ ]:
a = 1000000 # 원금 (단위 : 원)
b = 0.03 # 이자율
result = int(a * ((1 + b) ** 12))
print('1년 후 받는 원금 + 이자는 ' + str(result) + '원입니다.')

1년 후 받는 원금 + 이자는 1425760원입니다.


**문제 9**

문자열 연산을 이용하여 다음의 실행 결과가 나오도록 코드를 
다양한 방식으로 작성하여라

- 경우 1

    ```
    안녕하세요. 안녕하세요. 안녕하세요.
    ```

- 경우 2

    ```
    안녕하세요.
    안녕하세요.
    안녕하세요.
    안녕하세요.
    안녕하세요.
    ```

**문제9 답안**

In [ ]:
a = '안녕하세요. '
print(a + a + a)
print(a * 3)
print("안녕하세요. 안녕하세요. 안녕하세요.")

안녕하세요. 안녕하세요. 안녕하세요. 
안녕하세요. 안녕하세요. 안녕하세요. 
안녕하세요. 안녕하세요. 안녕하세요.


In [ ]:
b = '안녕하세요.\n안녕하세요.\n안녕하세요.\n안녕하세요.\n안녕하세요.'
c = '''안녕하세요.
'''
print(b)
print(a+'\n'+a+'\n'+a+'\n'+a+'\n'+a)
print(c+c+c+c+c)

안녕하세요.
안녕하세요.
안녕하세요.
안녕하세요.
안녕하세요.
안녕하세요. 
안녕하세요. 
안녕하세요. 
안녕하세요. 
안녕하세요. 
안녕하세요.
안녕하세요.
안녕하세요.
안녕하세요.
안녕하세요.



**문제 10**

`dogs`와 `cats` 두 개의 변수가 다음과 같이 선언되었다.

```python
dogs = '12'
cats = '7'
```

1. `dogs`와 `cats` 변수에 할당된 값의 자료형을 확인하여라.  
1. 강아지가 고양이보다 몇 마리 더 많은지 확인하는 코드를 작성하여라. 
    출력된 결과는 아래와 같은 형식이어야 한다.

    > 강아지가 고양이보다 n 마리 더 많다.

힌트: `int()` 함수 또는 `float()` 함수를 이용할 수 있다.

**문제10 답안**

In [ ]:
dogs = '12'
cats = '7'
type(dogs)

str

In [ ]:
type(cats)

str

In [87]:
int(dogs) - int(cats)

5

In [88]:
print('강아지가 고양이보다 ' + str(int(dogs) - int(cats)) + '마리 더 많다.')

강아지가 고양이보다 5마리 더 많다.


**문제 11**

1. 변수할당과 연산기호를 이용해 아래 내용을 논리식으로 표현하라.

    - `p1`: `3`과 `4`가 다른지 여부 판단<br>
    - `p2`: `"hello"`와 `"hi"`가 같은지 여부 판단<br>
    - `p3`: `2`가 `8`보다 작거나 같은지 여부 판단<br>
    - `p4`: `12`가 `20`보다 큰지 여부 판단<br><br>

1. 아래의 결과가 `True`인지 `False`인지 판단하는 논리식을 작성하라.

    - `p1 and p3`
    - `p2 and not p4`
    - `p3 or p4`

**문제11 답안**

In [106]:
p1 = 3 != 4
print(p1)

True


In [101]:
p2 = "hello" == "hi"
print(p2)

False


In [102]:
p3 = 2 <= 8
print(p3)

True


In [103]:
p4 = 12 > 20
print(p4)

False


In [107]:
p1 and p3

True

In [110]:
p2 and not p4

False

In [111]:
p3 or p4

True

**문제 12**

15, 28이 각각 짝수인지 홀수인지를 판별하는 코드를 작성하여라.
출력된 결과는 아래와 같은 형식이다.

> 15는 짝수인가요? False  
> 28은 짝수인가요? True

**문제12 답안**

In [117]:
if 15 % 2 == 0 : print('15는 짝수인가요?True')
else : print('15는 짝수인가요?False')
if 28 % 2 == 0 : print('28는 짝수인가요?True')
else : print('28는 짝수인가요?False')

15는 짝수인가요?False
28는 짝수인가요?True


**문제 13**

아래의 표는 어느 마트의 단위당 판매 가격이다.

 |제목|단위|가격|
 |:----:|:----:|----|
 |사과|1개|5,400원|
 |돼지고기|100g|2,480원|
 |달걀|10개|7,980원|
 |우유|1개|2,560원|
 |오이|1개|2,250원|

 A는 이 마트에서 사과 3개와 돼지고기 600g, 달걀 5개를 구매하려고 한다. 
 A는 총 비용을 구하는 프로그램을 작성하기 위해 아래와 같이 변수를 선언하였다.
 
```python
apple = '3'
pork = '600'
egg = '5'
```

위 변수들을 활용하여 
아래와 같은 형식으로 총 비용을 출력하는 프로그램을 작성하라.

> 사과 3개, 돼지고기 600g, 달걀 5개를 구매하는 총 비용은 OOO 원이다. 

**문제13 답안**

In [125]:
apple = '3'
pork = '600'
egg = '5'

result = int(3 * 5400 + 6 * 2480 + 7980 / 2)
print('사과 ' + apple + '개, 돼지고기 ' + pork + 'g, 달걀 ' + egg + '개를 구매하는 총 비용은 ' + str(result) + '원이다.')

사과 3개, 돼지고기 600g, 달걀 5개를 구매하는 총 비용은 35070원이다.


**문제 14**

반지름이 $r$인 구의 부피는 $\frac{4}{3} \pi r^3$ 이다. 반지름이 5인 구의 부피는 얼마인가?



**문제14 답안**

In [128]:
r = 5
vol = 4/3 * 3.14 * (r ** 3)
print(vol)

523.3333333333334


**문제 15**

책의 정가는 권당 24,950원이지만, 각 권당 10% 의 할인을 받는다고 가정하자.
책 한 권의 운송료는 2,000원 이고, 두 권째 부터는 권당 500원을 추가로 내야 한다.
60권을 사서 배송을 받고자 하면 얼마를 지불해야 하는가?

**문제15 답안**

In [129]:
book = 24950
total = 60
result = (book * total) * 90 / 100 + (2000 + 500 * (total-1)) 
print(result)

1378800.0


**문제 16**

오전 6시 52분에 집을 떠나서, 가볍게 (km당 8분 15초 소요) 1km를 뛰고, 
이어서 좀 빠르게 (km당 7분 12초 소요) 3km를 뛴 후, 다시 가볍게 1km를 마저 달려 집으로 돌아오면 
집에 도착하는 시간은 언제인가?

**문제16 답안**

In [137]:
s1 = 8 * 60 + 15
s2 = 7 * 60 + 12

result = s1 * 2 + s2 * 3
print(str(result // 60) + "분" + str(result % 60) + "초")

38분6초


오전 6시 52분 + 38분 6초 = 오전 7시 30분 6초

**문제 17**

만약 10km를 43분 30초 만에 달렸다면, 마일(mile)당 소요 시간은 얼마인가? 
시간당 마일로 계산한 평균 속도는 얼마인가? 
단, 1마일은 1.61 킬로미터로 계산한다.

**문제17 답안**

In [150]:
# 거리 = 속력 * 시간
t1 = (43 * 60 + 30) / 10 / 3600  # 1km 당 시간
v1 = 1 / t1 # 1km 당 속력
d1 = 1 / v1 * 1.61 # 마일 당 시간
print(str(int((d1 * 3600) // 60)) + '분' + str(int((d1 * 3600) % 60)) + '초')

7분0초


In [152]:
1 / d1

8.567144998929107

**문제 18**

1부터 10까지의 합을 계산하는 코드를 `for` 반복문을 이용하여 구현하라.

힌트: 리스트를 활용한다.

In [168]:
n = [1,2,3,4,5,6,7,8,9,10]
sum = 0
for i in n:
  sum = sum + i
print(sum)

55
